core/src/cluster_info.rs

```diff
@@ -1756,11 +1758,18 @@ impl ClusterInfo {
                             .unwrap()
                             .set_shred_version(entrypoint.shred_version);
                         self.insert_self();
-                        *self.entrypoint.write().unwrap() = Some(entrypoint);
-                        *adopt_shred_version = false;
+                        *entrypoint_processed = true;
                     }
+                } else {
+                    *entrypoint_processed = true;
                 }
+
+                // Update the entrypoint's id so future entrypoint pulls correctly reference it
+                *self.entrypoint.write().unwrap() = Some(entrypoint);
```

Comment:
Unrelated to this pull-request (as it seems pre-existing logic), but seems like we are expecting the entry-point to never update its contact-info (and in particular its gossip address).
Is that a reasonable assumption? or, should we make an effort to sync `.entrypoint` if we get an updated contact-info?

Reply:
I think it's a reasonable assumption since the only time that changes is during a network restart (assuming the entrypoint is configured correctly)